packages/imperative/src/imperative/src/plugins/utilities/NpmFunctions.ts

```diff
@@ -119,3 +73,79 @@ export async function getPackageInfo(pkgSpec: string): Promise<{ name: string, v
         return pacote.manifest(pkgSpec);
     }
 }
+
+export class NpmRegistryUtils {
+    /**
+     * Get the registry to install to.
+     * @param userRegistry Registry override specified on the command line
+     * @return {string}
+     */
+    public static getRegistry(userRegistry?: string): string {
+        if (userRegistry != null) return userRegistry;
+        const execOutput = ExecUtils.spawnAndGetOutput(npmCmd, ["config", "get", "registry"]);
+        return execOutput.toString().replace("\n", "");
+    }
+
+    /**
+     * NPM login to be able to install from secure registry
+     * @param {string} registry The npm registry to install from.
+     */
+    public static npmLogin(registry: string) {
+        ExecUtils.spawnAndGetOutput(npmCmd,
+            [
+                "login",
+                "--registry", registry,
+                "--always-auth",
+                "--auth-type=legacy"
+            ], {
+                stdio: [0,1,2]
```

Comment:
curious if we should do `stdio: "inherit"` 😋 
source: https://nodejs.org/docs/latest-v22.x/api/child_process.html#optionsstdio

Reply:
Good idea, changed it since `inherit` seems much easier to read than a list of 3 numbers 😁